### ISSUE TITLE
Add attendance limit to Meetup

### ISSUE THREAD

**Issue by fleetwood** [fleetwood/homecoming_algo#9]
Every Meetup is limited to a certain number of attendees. The Model should support the ability to add/remove attendees, and refuse an add if the limit has been reached.

Pass requirements:
- Meetup class has a `capacity` property, which is an integer.
- Meetup class has an `attendees` property, which is an array of `Students`.
- Meetup class has an **Add Attendee** method
- Meetup class has a **Remove Attendee** method
- **Add Attendee** should not allow new attendees once capacity has been reached.

Unknowns:
- If capacity is reached, does the Model throw an error when subsequent attendees are added? How do we know if adding was successful or not?
- Do we need to account for this business rule in mocks?

**Comment by fleetwood** [comment]
One thing I forgot to add. We should make sure that the same person can't be added twice.

Pass requirement:
- No duplicates

**Comment by JLDenney** [comment]
So I create this array class in a separate file. Do I then create separate files for each model's version of the array or do I instantiate it at the top of the existing model's file? So the instructor.js file will have the instructor class, but also the instructorArray?

**Comment by fleetwood** [comment]
Definitely a separate file, but no you don't instantiate at top of model.  Just to be clear, when you say instantiate, I assume you don't mean require:
`const item = require('/path/to/item')`

I assume you mean *instantiate* which is also called *newing*, as in 
`const c = new Letter()`

So you might require the file, but you won't instantiate it until you need it in code.

I added the **ExtendedArray** class for you. Two reasons: 
1. I like to code, too
2. I didn't want just an InstructorArray, but a more generic Array that we could use in multiple classes.

The work isn't done in the **ExtendedArray** class though. I just got you started. Check it out.

**Comment by JLDenney** [comment]
Yeah, I knew you didn't want just an array for instructors. I meant you'd create the generic extendedArray, then at the top of, say, instructors, you'd make instructorArray. At the top of students would be studentArray, etc.
